rat/__init__.py

```diff
@@ -101,7 +103,146 @@
 )
 ```
 
-## Basics
+## Learning by Analogy
+
+Rat builds on syntax and technology pioneered in a few different areas. To make the
+connections clearer, we will call these connections out. Rat should be understood as
+its own thing (learning by analogy is likely to be ineffective), but it was also
+developed at a certain place and time and we believe the source material is enlightening.
+
+Rat's *sampling* statements come from Stan:
+
+```
+y' ~ normal(a * x + b, sigma);
+```
+
+This would compile in a Stan model -- the thing you should notice though is the single
+tick mark. In Stan that would mean transpose but in Rat that has to do with
+vectorization. In Rat as in Stan, sampling statements increment a hidden unormalized
+log density variable (known as `target` in Stan). The statements themselves do
+not actually lead to any sampling.
+
+Rat uses dataframes as input, and by referencing columns of these dataframes Rat
+programs can vectorize over them. This comes from the lm, lme4, brms, rstanarm world.
+
+For instance, the above Rat program could be coupled with the dataframe:
+
+```
+ y, x
+10, 1
+15, 2
+21, 3
+24, 4
+```
+
+In this case, the sampling statement will accumulate four terms into the unnormalized
+log density. In computing each term, the symbols `x` and `y` will be substituted with
+values from the dataframe.
+
+The same model could also be coupled with the following dataframe:
+
+```
+ y, x, sigma
+10, 1,   1.1
+15, 2,   2.0
+21, 3,   1.7
+24, 4,   0.9
+```
+
+In this case, the values for `sigma` would come from the dataframe as well.
+
+This may lead you to ask, in any case, where do the values for `a` and `b` come from
+(and `sigma` itself if we're using the first dataframe)?
+
+We can see in the original sampling statement that there is enough information to
+infer what `a`, `b`, and `sigma` are -- they are scalar parameters that we'll need
+to do MCMC on.
+
+How about something more complex? Let us estimate a logit scale win probability for a
+group of teams based on some win loss data:
+
+```
+made, player
+   0,  curry
+   0,  curry
+   1,  curry
+   1,  green
+   0,  thompson
+   1,  thomspon
+```
+
+```
+made' ~ bernoulli_logit(skill[player]);
+```
+
+In this case the syntax suggests that there is one `skill` parameter for each
+player somewhere. Rat treats unresolved parameters like this like functions and
+is in this way works similarly to Bean Machine.
```

Comment:
I would eventually support saying parameters that are uninitialized with a prior should be illegal in rat.